Add "configure" to install docs for RN, Android and iOS (#6409)
PostHog/posthog.com: contents/docs/integrate/_snippets/install-react-native.mdx

In your React Native or Expo project add the `posthog-react-native` package to your project as well as the required peer dependencies.

> #### Expo Apps

Reviewer: ```suggestion
#### Expo apps
```

```sh
expo install posthog-react-native expo-file-system expo-application expo-device expo-localization
```

> #### React Native Apps

Reviewer: ```suggestion
#### React Native apps
```

```sh
yarn add posthog-react-native @react-native-async-storage/async-storage react-native-device-info
# or
npm i -s posthog-react-native @react-native-async-storage/async-storage react-native-device-info
```

### Configuration

#### With the PosthogProvider

> The recommended flow for setting up PostHog React Native is to use the `PostHogProvider` which utilises the Context API to pass the PostHog Client around as well as enabling autocapture and ensuring that the queue is flushed at the right time:

Reviewer: ```suggestion
The recommended way to set up PostHog for React Native is to use the `PostHogProvider` which utilizes the Context API to pass the PostHog client around, enable autocapture, and ensure that the queue is flushed at the right time:
```

> ```jsx

Reviewer: ```suggestion
```js
```

// App.(js|ts)
import { usePostHog, PostHogProvider } from 'posthog-react-native'
...

export function MyApp() {
    return (
        <PostHogProvider apiKey="<ph_project_api_key>" options={{
            // (Optional) PostHog API host (https://app.posthog.com by default)
            host: '<ph_instance_address>',
        }}>
            <MyComponent />
        </PostHogProvider>
    )
}

// Now you can simply access posthog elsewhere in the app like so

const MyComponent = () => {
    const posthog = usePostHog()

    useEffect(() => {
        posthog.capture("MyComponent loaded", { foo: "bar" })
    }, [])
}
```

### Configuration

#### Without the PosthogProvider

Due to the Async nature of React Native, PostHog needs to be initialised asynchronously in order for the persisted state to be loaded properly. The `PosthogProvider` takes care of this under-the-hood but you can alternatively create the instance yourself like so:

```tsx
// posthog.ts
import PostHog from 'posthog-react-native'

export let posthog: PostHog | undefined = undefined

export const posthogAsync: Promise<PostHog> = PostHog.initAsync('<ph_project_api_key>', {

>   // PostHog API host (https://app.posthog.com by default)
>   host: '<ph_instance_address>'

Reviewer: ```suggestion
  host: '<ph_instance_address>' // TIP: This is https://app.posthog.com by default
```

})

posthogAsync.then(client => {
  posthog = client
})

// app.ts
import { posthog, posthogAsync} from './posthog'

export function MyApp1() {
    useEffect(async () => {
        // Use posthog optionally with the possibility that it may still be loading
        posthog?.capture('MyApp1 loaded')
        // OR use posthog via the promise
        (await posthogAsync).capture('MyApp1 loaded')
    }, [])

    return <View>Your app code</View>
}

// You can even use this instance with the PostHogProvider
export function MyApp2() {
  return <PostHogProvider client={posthogAsync}>{/* Your app code */}</PostHogProvider>
}
```

### Configuration options

> For most people, the default configuration options will work great but if you need to you can further customise how PostHog will work.

Reviewer: ```suggestion
You can further customize how PostHog works through its configuration on initialization.
```
